### PR TITLE
Remove "up here" arrow on item-infos

### PR DESCRIPTION
**Pull request by jsha**
Use spacing to distinguish what is related to a given heading.

This was originally introduced in #53043, in response to #51387. The arrow is a little distracting, and leads the item-info to not be aligned properly with the text below it.

Demo: https://rustdoc.crud.net/jsha/impl-spacing/std/string/struct.String.html

r? @GuillaumeGomez

### CODE REVIEW

**Comment by Urgau** [comment]
I like it but its a bit weird on top item; maybe it's because it's so off to the right.
image

**Comment by jsha** [comment]
Updated to pass tests and pushed a fresh demo. Note that I had to delete the "We collapse it" part of the hash-item-expansion test. It wasn't actually working the way it was supposed to- it was working somewhat accidentally because the display of details toggles is broken in the version of puppeteer bundled with browser-UI-test, and the moving around of margins made it break without a good possibility of fixing. The functionality tested - that clicking a toggle opens and closes it - is already adequately tested in other test files, and doesn't need to be specifically tested in hash-item-expansion, which is about making sure items are _expanded_ when they're linked to.

**Comment by Urgau** [comment]
@jsha Regarding [my comment](https://github.com/rust-lang/rust/pull/92651#issuecomment-1007630153) you could simply add `margin-left: 0;` [here](https://github.com/rust-lang/rust/pull/92651/files#diff-67c3587d759aa9578993149e68eb8d3a9de6b28e5c0b85091b0a71a07f441795R770-R772) on `#main-content > .item-info`.

**Before**:
image

**After**:
image

**Comment by GuillaumeGomez** [comment]
I like @Urgau's suggestion. (if you apply it, please add a GUI test for it)

**Comment by jsha** [comment]
Done! Thanks for the suggestion, @Urgau.

File: src/test/rustdoc-gui/hash-item-expansion.goml [hunk 1]
```diff
@@ -4,10 +4,6 @@ goto: file://|DOC_PATH|/test_docs/struct.Foo.html#method.borrow
 assert-attribute: ("#blanket-implementations-list > details:nth-child(2)", {"open": ""})
 // We first check that the impl block is open by default.
 assert-attribute: ("#implementations + details", {"open": ""})
-// We collapse it.
```
Comment (GuillaumeGomez) [inline src/test/rustdoc-gui/hash-item-expansion.goml:7]: Why did you remove this?

**Comment by jsha** [comment]
I mention that in my last comment:

>  Note that I had to delete the "We collapse it" part of the hash-item-expansion test. It wasn't actually working the way it was supposed to- it was working somewhat accidentally because the display of details toggles is broken in the version of puppeteer bundled with browser-UI-test, and the moving around of margins made it break without a good possibility of fixing. The functionality tested - that clicking a toggle opens and closes it - is already adequately tested in other test files, and doesn't need to be specifically tested in hash-item-expansion, which is about making sure items are expanded when they're linked to.

**Review by GuillaumeGomez — APPROVED** [review]

**Comment by GuillaumeGomez** [comment]
Sorry, missed it. Then all good for me, thanks!

@bors: r+ rollup

**Comment by GuillaumeGomez** [comment]
Weird... Let's retry.

@bors: r+

**Comment by GuillaumeGomez** [comment]
cc @rust-lang/infra ^
